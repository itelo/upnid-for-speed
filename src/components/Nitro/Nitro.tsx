import React from "react";
import styled from "styled-components";
import { motion } from "framer-motion";
import Typography from "../Typography/Typography";

type NitroProps = {
  value: number;
};

const NitroBarContainer = styled.div`
  position: absolute;
  top: 25px;
  left: 25px;
  border: 5px solid black;
  width: 10rem;
  height: 2rem;
`;

const NitroBarContent = styled(motion.div)`
  background: blue;
  height: 100%;
`;

const Nitro = (props: NitroProps) => {
  return (
    <NitroBarContainer>
      <Typography style={{fontSize: '1.5rem', position: 'absolute', margin: '0.3rem 1rem'}} family="PressStart2P" color="yellow">Nitro</Typography>
      <NitroBarContent
        animate={{
          width: `${10 * props.value}%`
        }}
      />
    </NitroBarContainer>
  );
};

Nitro.defaultProps = {};

export default Nitro;
