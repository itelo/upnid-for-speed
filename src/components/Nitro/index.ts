export { default as Nitro } from './Nitro';
export * from './Nitro';