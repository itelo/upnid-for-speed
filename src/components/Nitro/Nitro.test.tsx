import React from "react";
import { render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import Nitro from "./Nitro";
import "jest-styled-components";

describe("<Nitro />", () => {
  it("test basic render", async () => {
    const { getByTestId } = render(<Nitro value={5} />);

    await waitFor(() =>
      expect(getByTestId("nitrobar-content")).toHaveStyle("width: 50%")
    );
  });

  it("test snapshot", () => {
    const { container } = render(<Nitro value={5} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });
});
