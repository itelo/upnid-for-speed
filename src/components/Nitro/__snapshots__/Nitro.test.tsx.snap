// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nitro /> test snapshot 1`] = `
.c1 {
  font-family: "Press Start 2P",cursive;
  color: rgb(234,199,133);
  font-size: 4.5rem;
  text-align: center;
  margin: 0;
}

.c0 {
  position: absolute;
  top: 25px;
  left: 25px;
  border: 5px solid black;
  width: 10rem;
  height: 2rem;
}

.c2 {
  background: blue;
  height: 100%;
}

<div
  class="c0"
  data-testid="nitrobar-container"
>
  <p
    class="c1"
    color="yellow"
    data-testid="nitro-text"
    style="font-size: 1.5rem; position: absolute; margin: 0.3rem 1rem;"
  >
    Nitro
  </p>
  <div
    class="c2"
    data-testid="nitrobar-content"
    style="width: 0px;"
  />
</div>
`;
